pkg/util/mount/exec_mount.go

```diff
@@ -0,0 +1,140 @@
+// +build linux
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+)
+
+// ExecMounter is a mounter that uses provided Exec interface to mount and
+// unmount a filesystem. For all other calls it uses a wrapped mounter.
+type execMounter struct {
+	wrappedMounter Interface
+	exec           Exec
+}
+
+func NewExecMounter(exec Exec, wrapped Interface) Interface {
```

Comment:
Is it intentional that this does not touches `SafeFormat` portions of work?

Reply:
`SafeFormat` is itself a wrapper over `mount.Interface`. It will work if it wraps `ExecMounter` instead of the generic Linux `Mounter`